tests/testthat/test-gls-models.R

```diff
@@ -181,3 +181,13 @@ test_that("New REML calculations work.", {
   check_REML2(Laski_hetAR1)
   check_REML2(Laski_CAR1)
 })
+
```

Comment:
Please add a few further tests, including for

* A model that has independent errors, so the only variance component is sigma^2. Check that the df should then be N - p, where N is the total number of observations.
* A model that has treatment-by-case interactions. Use the treatment effect for one of the cases in the numerator of the ES.